requests/__init__.py

```diff
@@ -48,11 +48,20 @@
 __copyright__ = 'Copyright 2016 Kenneth Reitz'
 
 # Attempt to enable urllib3's SNI support, if possible
-try:
-    from .packages.urllib3.contrib import pyopenssl
-    pyopenssl.inject_into_urllib3()
-except ImportError:
-    pass
+# If the user has a recent version of OpenSSL, this
+# isn't needed as Python's core ssl module handles SNI, etc correctly.
+# If the requirements of urllib3.contrib.pyopenssl change,
+# this version should be modified appropraitely.
+#
+# See https://github.com/kennethreitz/requests/issues/3213 as well.
+import ssl
+
+if ssl.OPENSSL_VERSION_INFO < (1, 0, 1):
```

Comment:
> Loading contrib.pyopenssl brings in a bunch of code I don't want, including pkg_resources via cryptography.

I didn't look at the code but it sounds like some package in the chain of imports (cryptography?) does too much work (probably finding/registering plugins) in the init module which in turn requires using pkg_resources. If it's the case then we have classic problem of unwanted side effects during import. The solution would be to either make this work explicit instead of implicit by requiring user of the library to request this functionality or to postpone work until it's really needed by introducing some laziness. The former makes the usage of library marginally harder and the latter is not always possible.